File: maven-confluence-core/src/test/java/org/bsc/confluence/rest/Issue213Test.java
package org.bsc.confluence.rest;

import org.junit.Test;

public class Issue213Test {

    @Test(expected = java.lang.NumberFormatException.class )
    public void bigIdWithInt() {

        String pageId = "2182288832";

        int result = Integer.valueOf( pageId );


    }

    @Test
    public void bigIdWithLong() {

        String pageId = "2182288832";

        long result = Long.valueOf( pageId );


    }
}

File: maven-confluence-core/src/test/kotlin/org/bsc/confluence/rest/Issue213Test.kt
package org.bsc.confluence.rest

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test


class Issue213Test {
    @Test
    fun bigIdWithInt() {

        assertThrows( NumberFormatException::class.java) {
            val pageId = "2182288832"
            val result = Integer.valueOf(pageId)
        }
    }

    @Test
    fun bigIdWithLong() {
        val pageId = "2182288832"
        val result = java.lang.Long.valueOf(pageId)

        assertEquals( 2182288832, result )
    }
}